src/store/mutations.js

```diff
@@ -517,6 +518,20 @@ function setTradesBucketed(state, apiBuckets) {
   state.tradeHistory.apiBuckets = apiBuckets;
 }
 
+function appendValuationsToMarktsForPrevewing(state, valuations) {
+  const lowercaseCurrency = settings.getCurrency().toLowerCase();
+
+  state.markets = state.markets.map((market) => {
```

Comment:
This could get the markets with uniq base currencies first, so it doesn't iterate the same `market.baseCurrency` repeatedly many times.

Reply:
Actually, to do that it would have to work differently; you can disregard this comment.